corehq/ex-submodules/dimagi/utils/tests/test_decorators.py

```diff
@@ -0,0 +1,46 @@
+import sys
+from io import StringIO
+
+from testil import Regex, eq, replattr
+
+from .. import profile
```

Comment:
just curious about using relative module iterator rather than using full paths. Is it better to use one over the other or use relative in certain scenarios?

Reply:
Good question! @millerdev always seems thoughtful about choices like this, and I'm also interested in those thoughts.

Reply:
TLDR: Use them when it's easy to tell where the imported module is located. A long string of dots is a sign of reaching too far.

My general guideline is to use relative imports when importing things that live within the same package/component context as the importing module. So for example, it's usually good to use relative imports within a single django app. On the other hand, it's not recommended to use relative imports when traversing many packages because it couples the relative locations of the importing and imported items.

I agree with the pros and cons in [this tutorial](https://realpython.com/absolute-vs-relative-python-imports/#pros-and-cons-of-relative-imports).

Good: `corehq/apps/mocha/urls.py`
```py
from .views import MochaView
```
Bad: `corehq/apps/badapple/models.py`
```py
# relative import from different django app
from ..app_manager.models import Application

# relative import from foreign package
from ...form_processor.utils.sql import fetchall_as_namedtuple
```

Reply:
thank you @millerdev